### PR TITLE
Wallet functions

### PR DESCRIPTION
**Pull request by murphyj8**
wif_to_bytes wallet function added.
TxIn to Output function added.

Both of these functions were required for commitment_tokens.

removed some dead-code & unhelpful comments.

fixed a minor issue in mock_interface.py (this will be helpful for unit testing without a need to communicate with bitcoin)

### CODE REVIEW

File: src/python/mod.rs [hunk 1]
```diff
@@ -144,6 +144,13 @@ pub fn py_sig_hash(
     Ok(bytes.into())
 }
 
+#[pyfunction(name = "wif_to_bytes")]
+pub fn py_wif_to_bytes(_py: Python, wif: &str) -> PyResult<PyObject>{
```
Comment (j-wilden) [inline src/python/mod.rs:148]: As this parameter is used in the function, it should be renamed to py (remove the underscore)

File: src/python/py_wallet.rs [hunk 2]
```diff
@@ -29,6 +29,17 @@ const MAIN_PUBKEY_HASH: u8 = 0x00;
 const TEST_PUBKEY_HASH: u8 = 0x6f;
 
 // TODO: note only tested for compressed key
+// Given a WIF, return bytes rather than SigningKey
+pub fn wif_to_bytes(wif: &str) -> Result<Vec<u8>>{
+    // Ok(private_key) => Ok(private_key),
```
Comment (j-wilden) [inline src/python/py_wallet.rs:34]: Does this need to be in here?  If not, delete the comment?

**Review by j-wilden — CHANGES_REQUESTED** [review]
See in-line comments

File: src/python/mod.rs [hunk 3]
```diff
@@ -144,6 +144,13 @@ pub fn py_sig_hash(
...
+    let key_bytes = wif_to_bytes(wif);
+    let bytes = PyBytes::new_bound(_py, &key_bytes.unwrap());
```
[the remark this answers is not in the document]
Comment (murphyj8) [reply src/python/mod.rs]: done. It's replaced with a '?' operator to allow the error to propagate to the caller.

File: src/python/mod.rs (hunk 1 above)
Comment (murphyj8) [reply src/python/mod.rs:148]: removed

Comment (murphyj8) [reply src/python/mod.rs:148]: not removed .. changed to py

File: src/python/py_wallet.rs (hunk 2 above)
Comment (murphyj8) [reply src/python/py_wallet.rs:34]: Did you mean the dead-code? If so removed.

**Review by j-wilden — APPROVED** [review]

**Pull request merged**